Include/pyintrinsics.h

```diff
@@ -0,0 +1,28 @@
+#ifndef Py_PYINTRINSICS_H
+#define Py_PYINTRINSICS_H
+
+/* Return the smallest integer k such that n < 2**k, or 0 if n == 0.
+ * Equivalent to floor(lg(x))+1.  Also equivalent to: bitwidth_of_type -
+ * count_leading_zero_bits(x)
+ */
+
+#if defined(__GNUC__)
+#define HAVE_BIT_LENGTH
+static inline int bit_length(unsigned long d) {
+    return d ? (8 * sizeof(unsigned long) - __builtin_clzl(d)) : 0;
```

Comment:
Unfortunately I don't think so. `__builtin_clzl(0)` is undefined or rather has different behavior on various processors.